src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildListener.java

```diff
@@ -109,20 +109,21 @@ public void onCompleted(AbstractBuild<?, ?> build, TaskListener listener) {
             return;
         }
         GHPullRequest pullRequest = cause.getPullRequest();
-        // Remove previous build data of the pull request from the build
-        for (Iterator<? extends Action> iter = build.getAllActions().iterator(); iter.hasNext();) {
-            Action action = iter.next();
-            if (action instanceof BuildData) {
-                BuildData buildData = (BuildData) action;
-
-                if (buildData.getLastBuiltRevision() != null
-                    && pullRequest.getHead().getSha().equals(buildData.getLastBuiltRevision())) {
+        String sha = pullRequest.getHead().getSha();
 
+        if (sha != null) {
```

Comment:
You could reduce the cyclomatic complexity greatly with a method like this
```
private void removeAction(List<Action> actions, String actionId) {
    if(actionId == null) {
        LOGGER.warning( "Pull request sha string is null. No previous actions removed.");
        return;
    }
    for (Iterator<? extends Action> iter = actions.iterator(); iter.hasNext(); ) {
        Action action = iter.next();
        if (isRemovableAction(action, actionId) ) {
              iter.remove();
              return;
        }
    }
}
```
You could use that If you really need the performance boost. 

Otherwise it could be put more concise this way:
```
private void removeAction(List<Action> actions, String actionId) {
    if(actionId == null) {
        LOGGER.warning( "Pull request sha string is null. No previous actions removed.");
        return;
    }
    for (Action action: actions) {
        if (isRemovableAction(action, actionId) ) {
              actions.remove(action);
              return;
        }
    }
}
```